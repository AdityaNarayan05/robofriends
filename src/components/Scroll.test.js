import React from 'react';
import { shallow } from 'enzyme';
import Scroll from './Scroll';

describe('Scroll', () => {
  it('should render Scroll component', () => {
    const wrapper = shallow(<Scroll />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render children components', () => {
    const mockChildren = <div>Child component</div>;
    const wrapper = shallow(<Scroll>{mockChildren}</Scroll>);
    expect(wrapper.contains(mockChildren)).toBe(true);
  });

  it('should have correct styles', () => {
    const wrapper = shallow(<Scroll />);
    const expectedStyles = {
      overflow: 'scroll',
      border: '5px solid black',
      height: '800px',
    };
    expect(wrapper.prop('style')).toEqual(expectedStyles);
  });
});
