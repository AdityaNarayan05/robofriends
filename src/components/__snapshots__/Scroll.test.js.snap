// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scroll should render Scroll component 1`] = `
<div
  style={
    Object {
      "border": "5px solid black",
      "height": "800px",
      "overflow": "scroll",
    }
  }
/>
`;
